src/validation.cpp

```diff
@@ -1179,9 +1217,10 @@ bool MemPoolAccept::Finalize(const ATMPArgs& args, Workspace& ws)
                 entry->GetTxSize(),
                 entry->GetFee()
         );
-        ws.m_replaced_transactions.push_back(it->GetSharedTx());
+        m_replaced_transactions.push_back(it->GetSharedTx());
     }
-    m_pool.RemoveStaged(ws.m_all_conflicting, false, MemPoolRemovalReason::REPLACED);
+    m_pool.RemoveStaged(m_all_conflicts, false, MemPoolRemovalReason::REPLACED);
+    m_all_conflicts.clear();
```

Comment:
fwiw that test is a single tx rbf, so even without this line it should pass. With package RBF it might result in multiple `RemoveStaged` calls on same conflicts, though I'm unsure what exactly that would do.

Reply:
Oh yes I was confused. Then I wonder why this is needed...?

Reply:
In package RBF contexts, the 2nd time `Finalize()` is called in a subpackage, the child tx, it will try and enumerate the `m_all_conflicts` list again, but those transactions were already removed, so we would crash.

Reply:
elaborated on the comment a bit more